sdk/core/azure-core/src/test/java/com/azure/core/implementation/jackson/AnimalWithTypeIdContainingDot.java

```diff
@@ -22,79 +20,36 @@ public abstract class AnimalWithTypeIdContainingDot implements JsonSerializable<
      * passed.
      */
     public static AnimalWithTypeIdContainingDot fromJson(JsonReader jsonReader) {
-        return fromJsonInternal(jsonReader, null);
-    }
-
-    static AnimalWithTypeIdContainingDot fromJsonInternal(JsonReader jsonReader,
-        String expectedODataType) {
         return JsonUtils.readObject(jsonReader, reader -> {
-            String odataType = null;
-            String breed = null;
-            boolean hasBreed = false;
-            Integer cuteLevel = null;
-            Integer tailLength = null;
-            List<String> meals = null;
+            // Buffer the JSON structure and make it replayable.
+            String json = JsonUtils.bufferedJsonObject(jsonReader);
```

Comment:
assuming the discriminator is the first, can we avoid buffering? It can be done later, for now it would be great to understand how the current API would evolve to allow it.

Reply:
Yes, an optimization can be done if the first property is the discriminator. Maybe this is a case where the first value can be peeked